api/src/main/java/net/wesjd/anvilgui/version/VersionProvider.java

```diff
@@ -0,0 +1,59 @@
+package net.wesjd.anvilgui.version;
+
+
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.bukkit.Bukkit;
+
+final class VersionProvider {
+    private static final Pattern VERSION_PATTERN =
+            Pattern.compile("(?i)\\(MC: (\\d)\\.(\\d+)\\.?(\\d+?)?(?: (Pre-Release|Release Candidate) )?(\\d)?\\)");
+
+    private final int major;
+    private final int minor;
+    private final int patch;
+
+    VersionProvider() {
+        this(Bukkit.getVersion());
+    }
+
+    VersionProvider(final String bukkitVersion) {
+        final Matcher matcher = VersionProvider.VERSION_PATTERN.matcher(bukkitVersion);
+        int major = 1;
+        int minor = 0;
+        int patch = 0;
+        if (matcher.find()) {
```

Comment:
Maybe I should put a warning or throw an exception to notify the user that using the library in completely different environment.

Reply:
I don't know, are there any other "Environments" that use Bukkit and don't follow the `VERSION_PATTERN`?

Reply:
that's the thing I am not aware of.

Reply:
Yeah, I think an exception should be thrown if it can't match. We need to give the devs/end users as much info as possible to debug their issue.